### DISCUSSION TITLE
How to set the maximum number of players?

### DISCUSSION THREAD

**Discussion by Lunasciel** [Minestom/Minestom#1827] [Help] [answered]
How to set the maximum number of players?

**Comment by ZakShearman** [comment, marked as answer]
Minestom has no built in way to do this. Instead, you should manage the maximum yourself and kick a player on the AsyncPlayerPreLoginEvent if there isn't any remaining space.

You can send the maximum for the server list using the ServerListPingEvent.

**Reply by dev-hydrogen** [reply]
This approach is also better for many reasons, since it simplifies a lot of stuff such as queue systems, overflow limits for staff, vip's etc.
